src/names/common.h

```diff
@@ -477,4 +479,23 @@ class CNameCache
 
 };
 
+/* This is where we store the the result of name_new, once initiated
+   by users of the UI (currently ui-only). We store these as strings
+   so that we can create a UniValue JSON object and send to RPC 
+   name_firstupdate. This also gets written to the wallet as a
+   JSON string and loaded back as such. */
+struct NameNewReturn
+{
+    bool ok;
+    std::string err_msg;
+    std::string toaddress;
+    std::string hex;
+    std::string rand;
+    std::string data;
+};
+
+/* Here is where we store our pending name_firstupdates (see above)
+   while we're waiting for name_new to confirm. */
+extern std::map<std::string, NameNewReturn > pendingNameFirstUpdate;
```

Comment:
For this one, what do you think about this refactor strategy:
- move `pendingNameFirstUpdate` map to walletdb
- move reads, writes, exists checks on `pendingNameFirstUpdate` to methods in walletdb
- in the UI do the interaction through walletmodel (QT model) which wraps function calls to walletdb

Reply:
Yes, basically.  I think you should add the variable to `CWallet`, where for instance also `mapWallet` is (since it is very similar in spirit, IMHO).  Access should be through `CWallet`, not walletdb, though.  (The walletdb is the backing BDB, but from there you load it into memory just like you do now - except that you keep it in `CWallet` and ideally do all access to it through the wallet.  Do not access walletdb directly, let the wallet do it; at least that's how I think it works for transactions.)

I don't know how the UI typically interacts with the wallet, but I guess walletmodel is the correct way, yes.  (In general, I think you should just follow how the wallet transactions are stored and handled, for instance.)